pkg/handlers/v1/cloud_fetch.go

```diff
@@ -108,6 +117,46 @@ func (h *CloudFetchByHostnameHandler) Handle(ctx context.Context, input CloudAss
 	return extractOutput(assets), nil
 }
 
+// CloudFetchAllAssetsByTimeHandler defines a lambda handler for bulk fetching cloud assets known at specific point in time
+type CloudFetchAllAssetsByTimeHandler struct {
+	LogFn   domain.LogFn
+	StatFn  domain.StatFn
+	Fetcher domain.CloudAllAssetsByTimeFetcher
+}
+
+// Handle handles fetching cloud assets by hostname
+func (h *CloudFetchAllAssetsByTimeHandler) Handle(ctx context.Context, input CloudAssetFetchAllByTimestampParameters) (CloudAssets, error) {
+	logger := h.LogFn(ctx)
+
+	ts, e := time.Parse(time.RFC3339Nano, input.Timestamp)
+	if e != nil {
+		logger.Info(logs.InvalidInput{Reason: e.Error()})
+		return CloudAssets{}, InvalidInput{Field: "time", Cause: e}
+	}
+
+	if input.Count == nil {
+		e = errors.New("missing or malformed required parameter count")
+		logger.Info(logs.InvalidInput{Reason: e.Error()})
+		return CloudAssets{}, InvalidInput{Field: "count", Cause: e}
+	}
+
+	var offset uint
+	if input.Offset != nil {
+		offset = *input.Offset
```

Comment:
suggestion: There is a difference between an optional value in the JSON payload and pointer values on the struct. We usually only introduce pointer values when we need to distinguish between explicit zero and implicit zero. My understanding is that it does not matter for offset because implicit and explicit zero have the same meaning. I recommend, for this case, having the optional input value described in the JSON schema but having a non-pointer type field for the struct.

Reply:
Got it.